dds/CMakeLists.txt

```diff
@@ -632,9 +632,12 @@ opendds_target_sources(OpenDDS_Dcps
   ${opendds_target_sources_args}
 )
 
+# Used in OpenDDSConfig.h.in, always true for CMake builds
+set(OPENDDS_CONFIG_AUTO_STATIC_INCLUDES 1)
+
 get_target_property(gendir OpenDDS_Dcps OPENDDS_GENERATED_DIRECTORY)
-set(config_file "${gendir}/dds/DCPS/config.h")
-configure_file("${_OPENDDS_CMAKE_DIR}/config.h.in" "${config_file}")
```

Comment:
In my machine, this creates a `config.h` under the `$DDS_ROOT/build/dds/opendds_generated/dds/DCPS/` directory. I imagine this change would create `OpenDDSConfig.h` under a similar directory. But if the `configure` script is used, looks like `OpenDDSConfig.h` will be put under `$DDS_ROOT/dds/`.

Do we want to have this difference? Or should we always create the file under the same directory?

Reply:
@iguessthislldo is going to try to fix that (I think?)

Reply:
CMake and MPC have different models of where to put generated files. In CMake it use's (at least by convention) what's called "out-of-source". It places files a build directory of the user's choosing. This approach's main advantage is that multiple builds can be done from the same source tree. I should also mention that when OpenDDS is install using CMake, all the files go to the same place.

Reply:
I was wondering about the former case. I guess you're letting CMake handle the include path, so that should be no problem now.